src/app/public/modules/viewkeeper/viewkeeper.service.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  Injectable,
+  Optional
+} from '@angular/core';
+
+import {
+  SkyViewkeeper
+} from './viewkeeper';
+
+import {
+  SkyViewkeeperHostOptions
+} from './viewkeeper-host-options';
+
+import {
+  SkyViewkeeperOptions
+} from './viewkeeper-options';
+
+@Injectable()
+export class SkyViewkeeperService {
+
+  constructor(@Optional() private hostOptions?: SkyViewkeeperHostOptions) { }
+
+  public create(options: SkyViewkeeperOptions): SkyViewkeeper {
+    options = Object.assign({}, this.hostOptions || {}, options);
+
+    return new SkyViewkeeper(options);
+  }
+
+  public destroy(vk: SkyViewkeeper): void {
+    vk.destroy();
```

Comment:
If the consumer has access to the `SkyViewkeeper` instance (and can call `destroy()` from the instance themselves), is having the service method redundant?

Reply:
I would prefer it still go through the service, in case the service is keeping a tally of existing viewkeeper instances (which was at one point part of this implementation but was later removed). Since consumers will be SKY UX developers for the foreseeable future, I think it's something we can enforce among the team through convention.